atomix/cluster/src/test/java/io/atomix/raft/protocol/ControllableRaftServerProtocol.java

```diff
@@ -61,14 +66,18 @@ public ControllableRaftServerProtocol(
   public void receiveNextMessage() {
     final var rcvQueue = messageQueue.get(localMemberId);
     if (!rcvQueue.isEmpty()) {
-      rcvQueue.poll().getLeft().run();
+      final var msg = rcvQueue.poll();
+      msg.getLeft().run();
+      timeoutQueue.remove(msg.getRight());
     }
   }
 
   public void receiveAll() {
     final var rcvQueue = messageQueue.get(localMemberId);
     while (!rcvQueue.isEmpty()) {
-      rcvQueue.poll().getLeft().run();
+      final var msg = rcvQueue.poll();
```

Comment:
:wrench: this var name is confusing. It is a tuple which contains a runnable and future. This future we use to remove the timeout. I think we can improve here. Either by defining explicity mentioning the type or refine the name, at least indicating that it is a tuple and not a msg.